elastic/kibana
# [TableListView] Update ext API

As a last part of refactoring the existing `<TableListView />` component I wanted to make some changes to the public API (the props). 

* `tableColumns` to `tableColumn`: the new design can allow an extra column to be added for some apps (viz needs a "type" column). But we shouldn't allow any number of columns to be added as this will break the design. This also mean that the "tags" column that previously consumer had to provide is now added by default (if the `savedObjectsTagging` plugin is provided).

* `searchFilters`: removed. We don't need every consumer to declare the "Tags" search filter. Once they provide the `savedObjectsTagging` plugin the reusable component takes care of adding the filter

* `initialFilter`: removed. We will rely on the URL as the source of truth to set the filter (https://github.com/elastic/kibana/issues/135203).

## Better typings

The new component enforces that the saved objects fetched have a `title` attribute provided. We shouldn't add a link in the table on an `undefined` title.

## New `@kbn/content-management-table-list` package

Those are breaking changes for all the consumers of the `<TableListView />` component. I've decided to fork the component and move it to a package. I will focus the integration in the Dashboard app. There will be a last PR to integrate the new component into the other plugins (Maps, Graph, Viz and now Security that also displays a list of dashboards - https://github.com/elastic/kibana/pull/136221)

## How to test

The new "packaged" component has been integrated in the Dashboard list view. No new functionalities have been added so make sure that all existing functionalies work (create dashboard, edit, delete, tags appear).
